test/routes/assignment_cycle_process.js

```diff
@@ -0,0 +1,195 @@
+  const user3 = await create_models.create_user("3");
+
+  const room1 = await create_models.create_room();
+
+  const ru1 = await create_models.create_room_user(room1.id, user1.id);
+  const ru2 = await create_models.create_room_user(room1.id, user2.id);
+  const ru3 = await create_models.create_room_user(room1.id, user3.id);
+
+  const chore1 = await create_models.create_chore(
+    room1.id,
+    "Dishes",
+    5,
+    (recurring = true),
+    (upcoming = true)
+  );
+  const chore2 = await create_models.create_chore(
+    room1.id,
+    "Laundry",
+    3,
+    (recurring = false),
+    (upcoming = true)
+  );
+  const chore3 = await create_models.create_chore(
+    room1.id,
+    "Mow",
+    8,
+    (recurring = false),
+    (upcoming = true)
+  );
+  const chore4 = await create_models.create_chore(
+    room1.id,
+    "Clean House",
+    13,
+    (recurring = true),
+    (upcoming = true)
+  );
+  const chore5 = await create_models.create_chore(
+    room1.id,
+    "Scrub",
+    15,
+    (recurring = true),
+    (upcoming = false)
+  );
+
+  const preference11 = await create_models.create_preference(
+    chore1.id,
+    user1.id,
+    0
+  );
+  const preference12 = await create_models.create_preference(
+    chore1.id,
+    user2.id,
+    3
+  );
+  const preference13 = await create_models.create_preference(
+    chore1.id,
+    user3.id,
+    1
+  );
+
+  const preference21 = await create_models.create_preference(
+    chore2.id,
+    user1.id,
+    1
+  );
+  const preference22 = await create_models.create_preference(
+    chore2.id,
+    user2.id,
+    2
+  );
+  const preference23 = await create_models.create_preference(
+    chore2.id,
+    user3.id,
+    3
+  );
+
+  const preference31 = await create_models.create_preference(
+    chore3.id,
+    user1.id,
+    3
+  );
+  const preference32 = await create_models.create_preference(
+    chore3.id,
+    user2.id,
+    1
+  );
+  const preference33 = await create_models.create_preference(
+    chore3.id,
+    user3.id,
+    2
+  );
+
+  const preference41 = await create_models.create_preference(
+    chore4.id,
+    user1.id,
+    2
+  );
+  const preference42 = await create_models.create_preference(
+    chore4.id,
+    user2.id,
+    0
+  );
+  const preference43 = await create_models.create_preference(
+    chore4.id,
+    user3.id,
+    0
+  );
+
+  t.truthy(await PreferenceChecker.validateUserPreferencesByRoomId(room1.id));
+
+  const res1 = await request(app)
+    .post("/assignment/assignmentCycle")
+    .expect(200);
+
+  //Should have 4 assignments, 5 chores, 4 active, 2 upcoming,
+  const assignments1 = await Assignment.find({});
+  const activeAssignments1 = await Assignment.find({ active: true });
+  const chores1 = await Chore.find({});
+  const upcomingChores1 = await Chore.find({ upcoming: true });
+  const activeChores1 = await Chore.find({ active: true });
+
+  t.truthy(await PreferenceChecker.validateUserPreferencesByRoomId(room1.id));
```

Comment:
t.true(await PreferenceChecker.validateUserPreferencesByRoomId(room1.id)) is a stronger assertion than truthy